lib/whitehall_importer/import.rb

```diff
@@ -2,32 +2,33 @@
 
 module WhitehallImporter
   class Import
-    attr_reader :whitehall_document
+    attr_reader :document_import, :whitehall_document
 
     def self.call(*args)
       new(*args).call
     end
 
-    def initialize(whitehall_document)
-      @whitehall_document = whitehall_document
+    def initialize(document_import)
+      @document_import = document_import
+      @whitehall_document = document_import.payload
     end
 
     def call
       ActiveRecord::Base.transaction do
         user_ids = create_users(whitehall_document["users"])
-        document = create_document(user_ids)
+        document_import.update!(document: create_document(user_ids))
 
         whitehall_document["editions"].each_with_index do |edition, edition_number|
           CreateEdition.call(
-            document: document,
+            document_import: document_import,
             current: current?(edition),
             whitehall_edition: edition,
             edition_number: edition_number + 1,
             user_ids: user_ids,
           )
         end
 
-        document
+        document_import.document
```

Comment:
Do we use the return from this anywhere? I wonder if we need to return anything since all we doing is mutating the document_input model.

Reply:
There are 4 tests in import_spec.rb that fail if I remove this line. I agree it (and the tests) can probably be removed, but feels out of scope for this PR.

Reply:
Sure, we can put together a list of things to do post merge?